litebird_sim/simulations.py

```diff
@@ -322,52 +322,46 @@ def flush(self):
 
     def create_observations(
         self,
-        detectors: List[Detector],
+        detectors: list,
         num_of_obs_per_detector: int,
         start_time,
         duration_s: float,
+        sampfreq_hz: float,
         distribute=True,
         use_mjd=False,
     ):
         "Create a set of Observation objects"
 
-        observations = []
-        for detidx, cur_det in enumerate(detectors):
-            cur_sampfreq_hz = cur_det.sampfreq_hz
-            num_of_samples = cur_sampfreq_hz * duration_s
-            samples_per_obs = distribute_evenly(num_of_samples, num_of_obs_per_detector)
-
-            if isinstance(start_time, float):
-                cur_time = start_time
+        self.observations = []
+        num_of_samples = sampfreq_hz * duration_s
+        samples_per_obss = distribute_evenly(num_of_samples, num_of_obs_per_detector)
+        start_samples = np.cumsum([0]+samples_per_obss[:-1])
+
+        # Keep only one every comm.size starting at my rank
+        start_samples = start_samples[self.mpi_comm.rank::self.mpi_comm.size]
+        samples_per_obss = samples_per_obss[self.mpi_comm.rank::self.mpi_comm.size]
+
+        for start_sample, samples_per_obs in zip(start_samples, samples_per_obss):
+            span_s = sampfreq_hz * start_sample
+            if use_mjd:
+                cur_start_time = Time(mjd=start_time.mjd) + TimeDelta(seconds=span_s)
             else:
-                if use_mjd:
-                    cur_time = start_time
-                else:
-                    cur_time = start_time.mjd
-
-            for cur_obs_idx in range(num_of_obs_per_detector):
-                nsamples = samples_per_obs[cur_obs_idx].num_of_elements
-                cur_obs = Observation(
-                    detector=cur_det,
```

Comment:
With your modifications, the parameter `detector` in `Observation.__init__` changed into `detectors`